src/parser.rs

```diff
@@ -262,21 +275,40 @@ impl<'a> Parser<'a> {
         }
     }
 
-    fn primary(&mut self) -> Expression {
-        let token = self.advance();
+    fn primary(& mut self) -> Expression {
```

Comment:
I think the method 'primay' skipped `TokenType::ident(_)` token because it used multiple `advances` to consume the token.

This approach mistakenly consumes the identifier token before we even reach the `function_call` function, leading to an error when expecting an identifier.